### ISSUE TITLE
Different focus behavior when clicking the toolbar buttons

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#6336]
Splitting this out from #3003 

When clicking or pressing Enter on some buttons, for example the alignment buttons, focus stays on that button. Instead, when activating other buttons, for example bold/italic/strikethrough, focus is moved back to the editing area.

This is inconsistent: all buttons should behave the same. It's also different from what TinyMCE does in the classic editor, where focus is always moved back to the editing area.

**Comment by tofumatt** [comment]
Not really convinced this needs design feedback, and I agree with you; this is bad for accessibility but beyond that is bad usability. This should be consistent, and always return focus to the content.

**Comment by tofumatt** [comment]
I'm having a bit of trouble following the code path for these controls and where the reset happens.

It seems like the reset of focus is a by-product/side-effect of https://github.com/WordPress/gutenberg/blob/master/packages/editor/src/components/rich-text/index.js#L448, so I'm not exactly clear on how to get the alignment controls (which are located elsewhere with different `onChange` events) to focus the control after being clicked.

@iseulde or someone else with more RichText knowledge than me: any chance you'd know off the top-of-your-head how to get the alignment controls to focus the block's content after interaction?

**Comment by gziolo** [comment]
@ellatrix - is this something you have planned to work on?

**Comment by mtias** [comment]
Related https://github.com/WordPress/gutenberg/pull/18534

**Comment by youknowriad** [comment]
Related #17617

**Comment by joedolson** [comment]
This is an interesting issue to look at. I think that what we really have is a divide between what the expected behavior is for a button based on what it does: *design* options are not expected to move focus, and *markup* options are. 

In many block toolbars, the alignment is the only design feature present. Some blocks, like images, also have duotone; but mostly it's some form of alignment. (Wide format is basically just a variant on alignment, as are cover block content positioning, etc.)

I think there's a sound logic to this consistency, but I can also see how it might not seem obvious to users. At this point, with 7 years of history to behaving this way, I'm not sure it would make sense to change it. I'd want to see some solid user testing to test the differences with keyboard & screen reader users to determine whether there's any real advantage.
